honeycombio/resource_dataset_test.go

```diff
@@ -8,11 +8,16 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	honeycombio "github.com/honeycombio/terraform-provider-honeycombio/client"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestAccHoneycombioDataset_basic(t *testing.T) {
-	testDataset := testAccDataset()
+
+	testDatasetName := testAccDataset()
+	testDataset := honeycombio.Dataset{
```

Comment:
Rather than using the test dataset here I think it makes sense to make a _new_ dataset and then validate its attributes